amon/templates/install/agent.sh

```diff
@@ -99,16 +99,22 @@ function add_repos(){
 function install_agent() {
 
     # Install the necessary package sources
-    if [ $DISTRO == 'rpm' ]; then
+    if [ "$DISTRO" == "rpm" ]; then
         printf "\033[92m* Installing the Amon Agent package for RPM distros\n\033[0m\n"
 
         $sudo_cmd yum -y install amonagent
 
-    elif [ $DISTRO == 'debian' ]; then
+    elif [ "$DISTRO" == "debian" ] && ([ "$ARCHITECTURE" == "x86_64" ] || [ "$ARCHITECTURE" == "i686" ]); then
         printf "\033[92m\n* Installing the Amon Agent package for Debian distros\n\033[0m\n"
 
         $sudo_cmd apt-get install -y --force-yes amonagent
 
+        #Added in reply to https://github.com/amonapp/amon/issues/170
+        ARCHITECTURE=`uname -m`
```

Comment:
Why is there a check above for the ARCHITECTURE yet it isn't set until here?

This is causing installs to not work at all.